relationalai-sdk-snowflake/test/test_integration_snowflake.py

```diff
@@ -0,0 +1,292 @@
+        api.create_engine(self.session, self.engine_name)
+
+        list_engines_res = api.list_engines(self.session).collect()
+        self.assertTrue(len(list_engines_res) > 0)
+
+        first_engine = list_engines_res[0]
+
+        for field in self.ENGINE_RESPONSE_FIELDS:
+            self.assertTrue(hasattr(first_engine, field))
+
+    def test_get_engine(self):
+        api.create_engine(self.session, self.engine_name)
+
+        get_engine_res = api.get_engine(self.session, self.engine_name).collect()
+        self.assertTrue(len(get_engine_res) == 1)
+
+        first_engine = get_engine_res[0]
+
+        for field in self.ENGINE_RESPONSE_FIELDS:
+            self.assertTrue(hasattr(first_engine, field))
+
+    def test_use_engine(self):
+        print("test_use_engine", self.engine_name)
+        api.create_engine(self.session, self.engine_name)
+
+        use_engine_res = api.use_engine(self.session, self.engine_name)
+        self.assertEqual(use_engine_res[0][0], self.engine_name)
+
+        current_engine_res = api.get_current_engine(self.session).collect()
+        self.assertEqual(current_engine_res[0]["CURRENT_ENGINE"], self.engine_name)
+
+        with self.assertRaises(Exception):
+            api.use_engine(self.session, "random-engine-name-that-doesnt-exist")
+
+    def tearDown(self):
+        api.delete_engine(self.session, self.engine_name)
+        pass
+
+
+class TestExecApi(unittest.TestCase):
+    def setUp(self) -> None:
+        self.session = Session.builder.configs(connection_parameters).create()
+        self.engine_name = generate_engine_name()
+        self.db_name = generate_db_name()
+
+        api.use_schema(self.session, RAI_SCHEMA)
+        api.create_database(self.session, self.db_name)
+        api.create_engine(self.session, self.engine_name)
+
+    def test_exec(self):
+        res = json.loads(api.exec(self.session, self.db_name, self.engine_name, "def output = 1 + 1").collect()[0][0])
+
+        self.assertEqual(res[0][0], 2)
+
+    def tearDown(self) -> None:
+        api.delete_database(self.session, self.db_name)
+        api.delete_engine(self.session, self.engine_name)
+        self.session.close()
+
+
+class TestModelApi(unittest.TestCase):
+    def setUp(self) -> None:
+        self.session = Session.builder.configs(connection_parameters).create()
+        self.engine_name = generate_engine_name()
+        self.db_name = generate_db_name()
+
+        api.use_schema(self.session, RAI_SCHEMA)
+        api.create_database(self.session, self.db_name)
+        api.create_engine(self.session, self.engine_name)
+
+    def test_load_model_code(self):
+        res = api.load_model_code(self.session, self.db_name, self.engine_name, "my_model", "def mymax[x, y] = maximum[abs[x], abs[y]]")
+        res = json.loads(api.exec(self.session, self.db_name, self.engine_name, "def output = mymax[5, -10]").collect()[0][0])
+
+        self.assertEqual(res[0][0], 10)
```

Comment:
assuming its correct since I see elsewhere, but seems odd so asking, 
after calling collect()[0][0], are these index still there on the next call? (res[0][0])

Reply:
`.collect()` returns `List[Row]` and since the exec API in the snowflake integration service returns a stringified JSON when we do `.collect()[0][0]` we actually get the stringified JSON like `[ [10] ]` in this case, so in order to access the value I need to parse the JSON and use `[0][0]` again. 

It all comes to designing some result interface which we are still missing in the regular Python SDK. In case of Snowflake though, I feel like the result interface would be something dataframe-ish that would allow users not only have all the functionality of dataframes but also possibly use pandas on top of it.